### ISSUE TITLE
Unable to define specs for Field 32 with binary prefix and BCD data

### ISSUE THREAD

**Issue by rohit-zenda** [moov-io/iso8583#132]
I am unable to unpack ISO message which follows (BCD + EBCDIC) specs. 

The ISO Message as byte array is - 
`[2 0 50 96 100 129 8 240 128 54 0 0 0 0 0 0 0 17 0 9 34 7 3 64 97 0 0 0 73 17 22 66 35 8 64 0 0 0 6 87 96 35 241 242 246 245 240 247 244 249 241 241 241 246 249 245 247 242 245 240 243 244 230 243 153 227 164 194 135 169 246 199 148 216 169 210 245 230 137 149 163 133 153 134 133 147 147 64 194 129 131 146 64 212 129 153 146 133 163 64 211 227 64 211 193 210 197 64 195 201 227 64 64 228 227 228 226 5 64 64 64 64 241 8 64 10 244 249 240 240 240 248 244 241 242 241 6 0 0 0 0 0 0 16 64 0 0 0 0 0 0 0 9 89 131 3 133 73 66 51 5 128 0 0 0 2] `

The partial specs defined so far are - 
```
	spec := &iso8583.MessageSpec{
		Fields: map[int]field.Field{
			0: field.NewString(&field.Spec{
				Length:      4,
				Description: "Message Type Indicator",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			1: field.NewBitmap(&field.Spec{
				Description: "Bitmap",
				Enc:         encoding.Binary,
				Pref:        prefix.Binary.Fixed,
			}),
			3: field.NewString(&field.Spec{
				Length:      6,
				Description: "Processing Code",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			4: field.NewString(&field.Spec{
				Length:      12,
				Description: "Amount, Transaction",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			7: field.NewString(&field.Spec{
				Length:      10,
				Description: "Transmission Date and Time",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			10: field.NewString(&field.Spec{
				Length:      8,
				Description: "Conversion Rate, Cardholder Billing",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			11: field.NewString(&field.Spec{
				Length:      6,
				Description: "System Trace Audit Number",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			18: field.NewString(&field.Spec{
				Length:      4,
				Description: "Merchant Type",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			19: field.NewString(&field.Spec{
				Length:      3,
				Description: "Acquiring Institution Country Code",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
				Pad:         padding.Left('0'),
			}),
			22: field.NewString(&field.Spec{
				Length:      4,
				Description: "Point-of-Service Entry Mode Code",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			25: field.NewString(&field.Spec{
				Length:      2,
				Description: "POS Condition Code",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.Fixed,
			}),
			32: field.NewString(&field.Spec{
				Length:      11,
				Description: "Acquiring Institution Identification Code",
				Enc:         encoding.BCD,
				Pref:        prefix.BCD.LL,
				Pad:         padding.Left('0'),
			}),
		},
	}

```

**Comment by rohit-zenda** [comment]
@alovak  I think I found the intermittent issue with failure of current specs. 
The BCD Decoder looks at the next BCD as well even if it is not needed and length of source is longer than destination. 
I raised an issue [here](https://github.com/yerden/go-util/issues/2), but not sure if the library is being regularly maintained.

**Comment by alovak** [comment]
Solved in #137
